### ISSUE TITLE
[Bug] NaviCube does not respond to MSAA settings in preferences.

### ISSUE THREAD

**Issue by obelisk79** [FreeCAD/FreeCAD#6689]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=34&t=67442

### Version

0.20 (Development)

### Full version info

Unable to paste the full version info.

I have tested on Linux and Windows and using several weekly builds from the last 6 months 
to verify this issue using both a laptop and desktop computer.

### Issue description

In the UI on 1080p screens it is readily apparent that the NAVICUBE does not respond to the MSAA settings in preferences. This renders the face text difficult to read at off-angles and the unrefined edges of the navicube are incongruent with the more refined appearance of the rest of the UI.

![image](https://user-images.githubusercontent.com/4140247/165527571-f90ee8dd-baef-4251-891f-5c8a6f8272e6.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by obelisk79** [comment]
So, I did some experimenting after looking up the QPainter QT5 class, and there is the setRenderHints() method which seems to have antialiasing options.

However, when I tried adding the following at line 356 just below paint.begin(&image);

QPainter paint.setRenderHints(QPainter::Antialiasing, true) 

I don't see any anti-aliasing being applied. is there a better approach for this?

**Comment by donovaly** [comment]
This is a known issue and also in FreeCAD 0.19

**Comment by luzpaz** [comment]
![Screenshot_20220427_094704](https://user-images.githubusercontent.com/4140247/165533471-1c42c505-71f6-45f8-af4a-2b5f06559ed6.png)
Kinda looks the same on @realthunder's branch as well.

**Comment by realthunder** [comment]
@luzpaz Did you turn on MSAA. Increasing border width will make the effect more obvious. Just compare it with the axis line. The border of the cube is rendered as texture in upstream if I remembered correctly, which is why it does not respond to MSAA setting. I have changed it to OpenGL line rendering.

https://user-images.githubusercontent.com/1207888/165647225-a8963957-885b-47af-9c81-b0e6bde0d11c.mp4
